### PR TITLE
Maya2022 Update

### PR DESCRIPTION
**Pull request by Thingvold**
Maya changed to Python 3 in version 2022.  Most  changes are related to code upgrades (print statements and some FileIO methods)

Some additional changes:

Upgraded to QAvatar with upper arm roll bones and extra clavicle.  Sample data updated for this.

Removed special case Y up local rotations for backward compatibility with the original QAvatar.

### CODE REVIEW

**Comment by EmilRosenquist** [comment]
Make it clear that we need to use versions before this one to build the plugin for older versions of Maya since they don't support Python 3.

**Review by EmilRosenquist — APPROVED** [review]
Python 3 changes looks good, I'm not sure if all cases are covered when importing these different attributes on segments (Couplings, goals etc..). Can you ensure that those are correct @Henrik-Norgren

**Comment by Wyst3r** [comment]
@EmilRosenquist Attributes seems fine I think (I guess we don't care about backwards compatibility here?), only thing is that coupling DOF is assumed to be the same as the DOF being coupled (In QTM, RotationZ could in theory be coupled to e.g. RotationX). Is that intentional @Thingvold?

**Comment by Thingvold** [comment]
Making the assumption that the DOFs are the same axis avoids adding a lot of complexity to the attributes.  It's pretty rare to need it so I opted out of implementing it.

File: QExportSolver.py [hunk 1]
```diff
@@ -252,15 +253,78 @@ def _ExportDOFs(self, jointNode, level):
...
+                    self._Write( Spaces(level+3) + "</Couplings>")
+                self._Write( Spaces(level+2)+"</RotationY>")
+
+
             if bRZ: 
                 lb = math.radians(cmds.getAttr("%s.ZRotDoF_LowerBound" % nodeName))
                 ub = math.radians(cmds.getAttr("%s.ZRotDoF_UpperBound" % nodeName))
-                self._Write( Spaces(level+2)+"<RotationZ LowerBound=\""+str(lb)+"\" UpperBound=\""+str(ub)+"\"/>")
+                self._Write( Spaces(level+2)+"<RotationZ>")
+                self._Write( Spaces(level+3)+"<Constraints LowerBound=\""+str(lb)+"\" UpperBound=\""+str(ub)+"\"/>")
```
Comment (Henrik-Norgren) [inline QExportSolver.py]: Constraints -> Constraint

File: QExportSolver.py [hunk 2]
```diff
@@ -252,15 +253,78 @@ def _ExportDOFs(self, jointNode, level):
...
+                        self._Write( Spaces(level+4)+ "<Coupling Coefficient=\""+str(bCP2_coeff)+"\" DegreeOfFreedom=\"RotationX\" Segment=\""+ bCP2_segment+"\"/>")
+
+                    self._Write( Spaces(level+3) + "</Couplings>")
+                self._Write( Spaces(level+2)+"</RotationX>")
+
             if bRY:     
                 lb = math.radians(cmds.getAttr("%s.YRotDoF_LowerBound" % nodeName))
                 ub = math.radians(cmds.getAttr("%s.YRotDoF_UpperBound" % nodeName))
-                self._Write( Spaces(level+2)+"<RotationY LowerBound=\""+str(lb)+"\" UpperBound=\""+str(ub)+"\"/>")
+                #self._Write( Spaces(level+2)+"<RotationY LowerBound=\""+str(lb)+"\" UpperBound=\""+str(ub)+"\"/>")
```
Comment (Henrik-Norgren) [inline QExportSolver.py]: Any reason to keep the commented code?

**Review by Henrik-Norgren — CHANGES_REQUESTED** [review]

File: QExportSolver.py (hunk 2 above)
Comment (Thingvold) [reply QExportSolver.py]: No.

File: QExportSolver.py (hunk 1 above)
Comment (Thingvold) [reply QExportSolver.py]: Should be good now.

**Review by Henrik-Norgren — APPROVED** [review]

**Pull request merged**